All project in branch phersus

phersus opened the pull request:
# This is a nice Title

Be sure you are on the branch __you want__ your code to merge into.

Detailed description about what's in this branch.

The idea being to highlight and summarizes the changes done on this particular branch so Tom can have a fair idea of what this is about.

Commit:
Rebasing phersus with the latest develop.

All project in branch phersus

Tommy647 commented on .idea/.gitignore:
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
Did you mean to commit your IDE files?

Tommy647 commented on cmd/grpcServer/main.go:
```diff
@@ -31,7 +31,7 @@ const (
 func main() {
 	// create a tcp listener for your gRPC service
-	listener, err := net.Listen("tcp", "0.0.0.0:9090") //nolint:gosec // development use only
+	listener, err := net.Listen("tcp", "0.0.0.0:8088") //nolint:gosec // development use only
```
Ahh I clashed with your centos management page.

Do you think we should make this an ENV VAR so we can change port at runtime?

Tommy647 commented on grpc.proto:
```diff
@@ -11,7 +11,17 @@ message HelloRequest {
+  string Response = 1; // response string we use
+}
+
+message ByeRequest {
```
No comment describing this message?

Tommy647 commented on internal/grpcclient/client_test.go:
```diff
@@ -1,21 +1,27 @@
 package grpcclient
-
+/*
```
Please fix tests

Tommy647 commented on make/Makefile.util:
```diff
@@ -1,6 +1,6 @@
 ## open gRPC-UI a web based gRPC explorer
 grpcui:
-	@grpcui -plaintext localhost:9090
+	@grpcui -plaintext localhost:8088
```
We should look at passing this in as a variable?

Tommy647 reviewed (changes requested):
Good effort overall, couple of tiny comments

Commit:
Removing the IDE config file .idea

Commit:
Moving code around:

To have the type declaration at the top and the methods just below.
- greeter.hello
- dbgreeter.hello

Refactoring greeter struct into greet, for clarity.
- greeter.hello
- dbgreeter.hello

Commit:
Adding missing ctx verification to HelloServer.Bye method.

Tommy647 commented:
I did that on purpose ;) good practice for you

phersus commented:
What is what you did on purpose?
Approve the PR so I get the conflict notification?

On Fri, 12 Nov 2021 at 11:14, Tom Smith ***@***.***> wrote:

> I did that on purpose ;) good practice for you
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Tommy647/go_example/pull/2#issuecomment-967021173>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANEYJHCST3BWKF6WSZINSLTULTZJLANCNFSM5HXP6XQA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Phersus
Gustavo Orozco

Tommy647 commented:
Updated develop to cause the conflict

Pull develop 

Rebase your branch 

When and only when you happy it's working, push (with a force) your branch

phersus commented:
OK,
Not sure what should happen but this is what I've got:

You said, pull develop:

*[gorozco::chicky @ go_example] >> >> >> $ git checkout develop *
Switched to branch 'develop'
Your branch is up to date with 'origin/develop'.
[gorozco::chicky @ go_example] >> >> >> $ git pull origin
develop   HEAD      master    phersus
*[gorozco::chicky @ go_example] >> >> >> $ git pull *
remote: Enumerating objects: 71, done.
remote: Counting objects: 100% (71/71), done.
remote: Compressing objects: 100% (19/19), done.
remote: Total 45 (delta 23), reused 41 (delta 19), pack-reused 0
Unpacking objects: 100% (45/45), 10.12 KiB | 1.01 MiB/s, done.
From github.com:Tommy647/go_example
   971bd77..9ce94dd  develop    -> origin/develop
 * [new branch]      http-sync  -> origin/http-sync
Updating 971bd77..9ce94dd
Fast-forward
 README.md                          | 24 ++++++++++++++++++++----
 cmd/client/main.go                 |  3 ++-
 cmd/concurrent/main.go             | 63
++++++++++++++++++++++++++++++++++++++++++---------------------
 cmd/grpcServer/main.go             |  2 +-
 docker-compose.yaml                | 39
+++++++++++++++++++++++++++++++++++++--
 internal/dbgreeter/greeter.go      | 22 ++++++++++++----------
 internal/dbgreeter/greeter_test.go |  4 ++--
 internal/greeter/hello.go          | 14 +++++++-------
 internal/greeter/hello_test.go     |  4 ++--
 internal/grpcserver/server.go      | 12 ++++++------
 internal/grpcserver/server_test.go |  2 +-
 internal/httpserver/server.go      |  8 ++++----
 12 files changed, 136 insertions(+), 61 deletions(-)
*[gorozco::chicky @ go_example] >> >> >> $*

You said, rebase your branch:

*[gorozco::chicky @ go_example] >> >> >> $ git branch *
* develop
  master
  phersus
*[gorozco::chicky @ go_example] >> >> >> $ git checkout phersus *
Switched to branch 'phersus'
Your branch is ahead of 'origin/phersus' by 1 commit.
  (use "git push" to publish your local commits)
*[gorozco::chicky @ go_example] >> >> >> $ git rebase develop *
Auto-merging internal/grpcserver/server.go
Auto-merging docker-compose.yaml
CONFLICT (content): Merge conflict in docker-compose.yaml
Auto-merging cmd/grpcServer/main.go
error: could not apply d38550c... All project in branch phersus
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase
--abort".
Could not apply d38550c... All project in branch phersus
*[gorozco::chicky @ go_example] >> >> >> $*

Is this what was supposed to happen? Am I applying the right commands?
Should I do the git rebase --skip

If I'm understanding the rebase concept correctly, basically what we want
is to modify my branch phersus in a way so it is LIKE I created it AFTER
your last modification of develop, am I right?

Cheers,



On Fri, 12 Nov 2021 at 13:34, Tom Smith ***@***.***> wrote:

> Updated develop to cause the conflict
>
> Pull develop
>
> Rebase your branch
>
> When and only when you happy it's working, push (with a force) your branch
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Tommy647/go_example/pull/2#issuecomment-967122551>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANEYJHCVKDYIWGBI4SFNWX3ULUJXHANCNFSM5HXP6XQA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Phersus
Gustavo Orozco

Tommy647 commented:
Yup that all looks good. It'll stop so you can resolve the conflicts (change files, git add them and the git rebase --continue)

phersus commented:
Only problem is, I don't remember what to do :)

The issue is in the docker-compose file, where I see this:

<<<<<<< HEAD
    - "9090:9090"
    healthcheck:
      test:  nc -zvw10 grpc 9090
      interval: 1s
      timeout: 5s
      retries: 5
=======
    - "8088:8088"
>>>>>>> d38550c... All project in branch phersus

I think this is the only relevant bit I do care about when doing this
conflict resolution ... correct?
Not sure what I need to delete/keep tbh, my guess would be to make it like
this:

    - "8088:8088"
    healthcheck:
      test:  nc -zvw10 grpc 9090
      interval: 1s
      timeout: 5s
      retries: 5

then save the file and do the git add docker-compose.yaml then the git
rebase --continue.

Right?

On Fri, 12 Nov 2021 at 16:32, Tom Smith ***@***.***> wrote:

> Yup that all looks good. It'll stop so you can resolve the conflicts
> (change files, git add them and the git rebase --continue)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Tommy647/go_example/pull/2#issuecomment-967249477>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANEYJHCPGB357O3CO3KVTI3ULU6SNANCNFSM5HXP6XQA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Phersus
Gustavo Orozco

Tommy647 commented:
Yup,

you changed the ports, but I added some health checks (next to the port
line) which confused git

On Fri, 12 Nov 2021 at 17:02, phersus ***@***.***> wrote:

> Only problem is, I don't remember what to do :)
>
> The issue is in the docker-compose file, where I see this:
>
> <<<<<<< HEAD
> - "9090:9090"
> healthcheck:
> test: nc -zvw10 grpc 9090
> interval: 1s
> timeout: 5s
> retries: 5
> =======
> - "8088:8088"
> >>>>>>> d38550c... All project in branch phersus
>
> I think this is the only relevant bit I do care about when doing this
> conflict resolution ... correct?
> Not sure what I need to delete/keep tbh, my guess would be to make it like
> this:
>
> - "8088:8088"
> healthcheck:
> test: nc -zvw10 grpc 9090
> interval: 1s
> timeout: 5s
> retries: 5
>
> then save the file and do the git add docker-compose.yaml then the git
> rebase --continue.
>
> Right?
>
> On Fri, 12 Nov 2021 at 16:32, Tom Smith ***@***.***> wrote:
>
> > Yup that all looks good. It'll stop so you can resolve the conflicts
> > (change files, git add them and the git rebase --continue)
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/Tommy647/go_example/pull/2#issuecomment-967249477>,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ANEYJHCPGB357O3CO3KVTI3ULU6SNANCNFSM5HXP6XQA
> >
> > .
> > Triage notifications on the go with GitHub Mobile for iOS
> > <
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> >
> > or Android
> > <
> https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub
> >.
> >
> >
>
>
> --
> Phersus
> Gustavo Orozco
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Tommy647/go_example/pull/2#issuecomment-967270648>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHS36JZHYJSTCCUZZQ2LD3ULVCARANCNFSM5HXP6XQA>
> .
>

phersus commented:
Not sure this is going in the right direction ...
After removing the conflict with the docker-compose.yaml adding it and
continuing the rebase I've got some more conflicts, I'm guessing some are
because I refactored the Greeter struct into Greet.
Should I accept the version with Greeter coming in or should I retain the
Greet that I made locally? The others seemed to be changes in func names
you made?

This are the files still seeing conflicts:

dbgreeter/
    greeter.go
    greeter_test.go

greeter/
    hello.go
    hello_test.go

grpcserver/
    server_test.go





On Fri, 12 Nov 2021 at 17:34, Tom Smith ***@***.***> wrote:

> Yup,
>
> you changed the ports, but I added some health checks (next to the port
> line) which confused git
>
> On Fri, 12 Nov 2021 at 17:02, phersus ***@***.***> wrote:
>
> > Only problem is, I don't remember what to do :)
> >
> > The issue is in the docker-compose file, where I see this:
> >
> > <<<<<<< HEAD
> > - "9090:9090"
> > healthcheck:
> > test: nc -zvw10 grpc 9090
> > interval: 1s
> > timeout: 5s
> > retries: 5
> > =======
> > - "8088:8088"
> > >>>>>>> d38550c... All project in branch phersus
> >
> > I think this is the only relevant bit I do care about when doing this
> > conflict resolution ... correct?
> > Not sure what I need to delete/keep tbh, my guess would be to make it
> like
> > this:
> >
> > - "8088:8088"
> > healthcheck:
> > test: nc -zvw10 grpc 9090
> > interval: 1s
> > timeout: 5s
> > retries: 5
> >
> > then save the file and do the git add docker-compose.yaml then the git
> > rebase --continue.
> >
> > Right?
> >
> > On Fri, 12 Nov 2021 at 16:32, Tom Smith ***@***.***> wrote:
> >
> > > Yup that all looks good. It'll stop so you can resolve the conflicts
> > > (change files, git add them and the git rebase --continue)
> > >
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/Tommy647/go_example/pull/2#issuecomment-967249477
> >,
> > > or unsubscribe
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/ANEYJHCPGB357O3CO3KVTI3ULU6SNANCNFSM5HXP6XQA
> > >
> > > .
> > > Triage notifications on the go with GitHub Mobile for iOS
> > > <
> >
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> > >
> > > or Android
> > > <
> >
> https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub
> > >.
> > >
> > >
> >
> >
> > --
> > Phersus
> > Gustavo Orozco
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/Tommy647/go_example/pull/2#issuecomment-967270648>,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AAHS36JZHYJSTCCUZZQ2LD3ULVCARANCNFSM5HXP6XQA
> >
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Tommy647/go_example/pull/2#issuecomment-967292949>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANEYJHFZCMMIKF2B7SJ6FVDULVF4HANCNFSM5HXP6XQA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Phersus
Gustavo Orozco

Tommy647 commented:
You can decide which version you prefer 

Sent from my iPhone

> On 12 Nov 2021, at 18:08, phersus ***@***.***> wrote:
> 
> ﻿
> Not sure this is going in the right direction ...
> After removing the conflict with the docker-compose.yaml adding it and
> continuing the rebase I've got some more conflicts, I'm guessing some are
> because I refactored the Greeter struct into Greet.
> Should I accept the version with Greeter coming in or should I retain the
> Greet that I made locally? The others seemed to be changes in func names
> you made?
> 
> This are the files still seeing conflicts:
> 
> dbgreeter/
> greeter.go
> greeter_test.go
> 
> greeter/
> hello.go
> hello_test.go
> 
> grpcserver/
> server_test.go
> 
> 
> 
> 
> 
> On Fri, 12 Nov 2021 at 17:34, Tom Smith ***@***.***> wrote:
> 
> > Yup,
> >
> > you changed the ports, but I added some health checks (next to the port
> > line) which confused git
> >
> > On Fri, 12 Nov 2021 at 17:02, phersus ***@***.***> wrote:
> >
> > > Only problem is, I don't remember what to do :)
> > >
> > > The issue is in the docker-compose file, where I see this:
> > >
> > > <<<<<<< HEAD
> > > - "9090:9090"
> > > healthcheck:
> > > test: nc -zvw10 grpc 9090
> > > interval: 1s
> > > timeout: 5s
> > > retries: 5
> > > =======
> > > - "8088:8088"
> > > >>>>>>> d38550c... All project in branch phersus
> > >
> > > I think this is the only relevant bit I do care about when doing this
> > > conflict resolution ... correct?
> > > Not sure what I need to delete/keep tbh, my guess would be to make it
> > like
> > > this:
> > >
> > > - "8088:8088"
> > > healthcheck:
> > > test: nc -zvw10 grpc 9090
> > > interval: 1s
> > > timeout: 5s
> > > retries: 5
> > >
> > > then save the file and do the git add docker-compose.yaml then the git
> > > rebase --continue.
> > >
> > > Right?
> > >
> > > On Fri, 12 Nov 2021 at 16:32, Tom Smith ***@***.***> wrote:
> > >
> > > > Yup that all looks good. It'll stop so you can resolve the conflicts
> > > > (change files, git add them and the git rebase --continue)
> > > >
> > > > —
> > > > You are receiving this because you authored the thread.
> > > > Reply to this email directly, view it on GitHub
> > > > <https://github.com/Tommy647/go_example/pull/2#issuecomment-967249477
> > >,
> > > > or unsubscribe
> > > > <
> > >
> > https://github.com/notifications/unsubscribe-auth/ANEYJHCPGB357O3CO3KVTI3ULU6SNANCNFSM5HXP6XQA
> > > >
> > > > .
> > > > Triage notifications on the go with GitHub Mobile for iOS
> > > > <
> > >
> > https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> > > >
> > > > or Android
> > > > <
> > >
> > https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub
> > > >.
> > > >
> > > >
> > >
> > >
> > > --
> > > Phersus
> > > Gustavo Orozco
> > >
> > > —
> > > You are receiving this because you commented.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/Tommy647/go_example/pull/2#issuecomment-967270648>,
> > > or unsubscribe
> > > <
> > https://github.com/notifications/unsubscribe-auth/AAHS36JZHYJSTCCUZZQ2LD3ULVCARANCNFSM5HXP6XQA
> > >
> > > .
> > >
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/Tommy647/go_example/pull/2#issuecomment-967292949>,
> > or unsubscribe
> > <https://github.com/notifications/unsubscribe-auth/ANEYJHFZCMMIKF2B7SJ6FVDULVF4HANCNFSM5HXP6XQA>
> > .
> > Triage notifications on the go with GitHub Mobile for iOS
> > <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> > or Android
> > <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
> >
> >
> 
> 
> -- 
> Phersus
> Gustavo Orozco
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.

phersus commented:
Did you introduce some error in the code on purpose?
I'm confused whether those are problems of my refactoring or you added them
for us to fix them ...

I have fixed it, at least for me it was not working, now it is, but I'm not
sure if what I did was intended as an exercise or I did some bad
manipulation and now it's working ...

If this was intended then I'm happy I've found a solution (maybe not the
best one though), but if it is not ... maybe I'm missing something.

I will push my branch now!




On Fri, 12 Nov 2021 at 18:53, Tom Smith ***@***.***> wrote:

> You can decide which version you prefer
>
> Sent from my iPhone
>
> > On 12 Nov 2021, at 18:08, phersus ***@***.***> wrote:
> >
> > ﻿
> > Not sure this is going in the right direction ...
> > After removing the conflict with the docker-compose.yaml adding it and
> > continuing the rebase I've got some more conflicts, I'm guessing some are
> > because I refactored the Greeter struct into Greet.
> > Should I accept the version with Greeter coming in or should I retain the
> > Greet that I made locally? The others seemed to be changes in func names
> > you made?
> >
> > This are the files still seeing conflicts:
> >
> > dbgreeter/
> > greeter.go
> > greeter_test.go
> >
> > greeter/
> > hello.go
> > hello_test.go
> >
> > grpcserver/
> > server_test.go
> >
> >
> >
> >
> >
> > On Fri, 12 Nov 2021 at 17:34, Tom Smith ***@***.***> wrote:
> >
> > > Yup,
> > >
> > > you changed the ports, but I added some health checks (next to the port
> > > line) which confused git
> > >
> > > On Fri, 12 Nov 2021 at 17:02, phersus ***@***.***> wrote:
> > >
> > > > Only problem is, I don't remember what to do :)
> > > >
> > > > The issue is in the docker-compose file, where I see this:
> > > >
> > > > <<<<<<< HEAD
> > > > - "9090:9090"
> > > > healthcheck:
> > > > test: nc -zvw10 grpc 9090
> > > > interval: 1s
> > > > timeout: 5s
> > > > retries: 5
> > > > =======
> > > > - "8088:8088"
> > > > >>>>>>> d38550c... All project in branch phersus
> > > >
> > > > I think this is the only relevant bit I do care about when doing this
> > > > conflict resolution ... correct?
> > > > Not sure what I need to delete/keep tbh, my guess would be to make it
> > > like
> > > > this:
> > > >
> > > > - "8088:8088"
> > > > healthcheck:
> > > > test: nc -zvw10 grpc 9090
> > > > interval: 1s
> > > > timeout: 5s
> > > > retries: 5
> > > >
> > > > then save the file and do the git add docker-compose.yaml then the
> git
> > > > rebase --continue.
> > > >
> > > > Right?
> > > >
> > > > On Fri, 12 Nov 2021 at 16:32, Tom Smith ***@***.***> wrote:
> > > >
> > > > > Yup that all looks good. It'll stop so you can resolve the
> conflicts
> > > > > (change files, git add them and the git rebase --continue)
> > > > >
> > > > > —
> > > > > You are receiving this because you authored the thread.
> > > > > Reply to this email directly, view it on GitHub
> > > > > <
> https://github.com/Tommy647/go_example/pull/2#issuecomment-967249477
> > > >,
> > > > > or unsubscribe
> > > > > <
> > > >
> > >
> https://github.com/notifications/unsubscribe-auth/ANEYJHCPGB357O3CO3KVTI3ULU6SNANCNFSM5HXP6XQA
> > > > >
> > > > > .
> > > > > Triage notifications on the go with GitHub Mobile for iOS
> > > > > <
> > > >
> > >
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> > > > >
> > > > > or Android
> > > > > <
> > > >
> > >
> https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub
> > > > >.
> > > > >
> > > > >
> > > >
> > > >
> > > > --
> > > > Phersus
> > > > Gustavo Orozco
> > > >
> > > > —
> > > > You are receiving this because you commented.
> > > > Reply to this email directly, view it on GitHub
> > > > <
> https://github.com/Tommy647/go_example/pull/2#issuecomment-967270648>,
> > > > or unsubscribe
> > > > <
> > >
> https://github.com/notifications/unsubscribe-auth/AAHS36JZHYJSTCCUZZQ2LD3ULVCARANCNFSM5HXP6XQA
> > > >
> > > > .
> > > >
> > >
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/Tommy647/go_example/pull/2#issuecomment-967292949
> >,
> > > or unsubscribe
> > > <
> https://github.com/notifications/unsubscribe-auth/ANEYJHFZCMMIKF2B7SJ6FVDULVF4HANCNFSM5HXP6XQA
> >
> > > .
> > > Triage notifications on the go with GitHub Mobile for iOS
> > > <
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> >
> > > or Android
> > > <
> https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub
> >.
> > >
> > >
> >
> >
> > --
> > Phersus
> > Gustavo Orozco
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub, or unsubscribe.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Tommy647/go_example/pull/2#issuecomment-967341505>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANEYJHDY2XBJDUL3I47AKKLULVPBFANCNFSM5HXP6XQA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Phersus
Gustavo Orozco

Commit:
Fixing problems with the GreetProvider{} interface not being satisfied, basically because it was
expecting structs implementing Greet() method, but they were implementing HelloGreet() instead.

I renamed the method for the one expected to satisfy the interface and this fixed the issue.

Commit:
Delete .idea file

Commit:
Adding a CoffeeService on the rpc server
Adding db support for the new CoffeeService:
 creating a table, coffee with two records
  espresso
  macchiato

Commit:
Start adding DB support into gRPC query for coffee service.
The gRPC query must carry a string specifying what source of data will be used: local string || DB

Commit:
Adding a coffee service, database migrations update to take into account the new service

Commit:
Debugging the DB responses for the new Coffee Service

Commit:
Trying to add the DB as part of the GreetCoffee method

Commit:
Adding DB support for CoffeeService
 -Via gRPC you specify what coffee you want and the source for data: DB || other
 -If you asked for an existing coffee from the DB the price will returned
 -If you asked for a coffee that is not in the DB or you don't want the source to be a DB, you get a free drink!

Commit:
Cleaning some of the files where db credentials and parameters got copied and in the end not needed.
Added a simple test function to test CoffeeGreet() method on internal/greeter/hello.go

Commit:
More cleaning ... deleting unnecessary comment on dbgreeter/greeter.go

Commit:
Adding TestGreet_CoffeeGreet() mocking the DB

Commit:
Some more testing for the CoffeeService

Commit:
Adding a coffee service over http with DB support

Commit:
Adding WithRole middleware for the httpserver -broken-

Commit:
Adding WithRole middlewar is breaking but access to DB is working and serving coffee.

Commit:
Adding WithRole middleware, depending on role (admin and db) the
HandleCoffee is triggered. Then the HandleCoffee verifies wheather the
barista role is set serve an espresso or a free coffee otherwise

Commit:
Eliminating to doble check the db role.

Commit:
Adding the last change on jwt.go after testing without barista or db
roles

Commit:
Merge branch 'phersus' of github.com:Tommy647/go_example into phersus

To align phersus on delata and chicky

Commit:
Adding a more test for the CoffeeGreet

Commit:
Cleaning a bit auth.go to avoid shadowing the err var, the main.go on
httpserver to handle the possible error when closing the dbConn
